pkg/route/routecomponenthelpers/routecomponentcontrollerhelpers.go

```diff
@@ -0,0 +1,133 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/openshift/library-go/pkg/certs"
+)
+
+// RouteComponentControllerHelper aggregates necessary interfaces for Custom Certs for routes functions.
+type RouteComponentControllerHelper struct {
+	secretLister  corev1listers.SecretLister
+	ingressClient configsetterv1.IngressInterface
+}
+
+// NewRouteComponentHelper creates new RouteComponentControllerHelper instance
+func NewRouteComponentHelper(secretLister corev1listers.SecretLister, ingressClient configsetterv1.IngressInterface) *RouteComponentControllerHelper {
+	return &RouteComponentControllerHelper{secretLister: secretLister, ingressClient: ingressClient}
+}
+
+// UpdateComponentRouteStatus searches the entries of the ingress.status.componentRoutes array for a componentRoute with a matching namespace and name.
+// If a matching componentRoute is found, the two objects are compared minus any of the conditions.lastTransactionTime entries. If all the fields
+// match, the entry is updated.
+// If no matching componentRoute is found, the entry is appended to the list.
+// true is returned if the status of the ingress.config.openshift.io/cluster is updated.
+func UpdateComponentRouteStatus(ctx context.Context, routeComponentHelper *RouteComponentControllerHelper, componentRoute *configv1.ComponentRouteStatus) (bool, error) {
+	// Override the timestamps
+	now := metav1.Now()
+
+	// Create a copy for comparison and remove transaction times
+	componentRouteCopy := componentRoute.DeepCopy()
+	setLastTransactionTime(componentRouteCopy, now)
+
+	updated := false
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		ingressConfig, err := routeComponentHelper.ingressClient.Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		existingComponentRoute := GetComponentRouteStatus(ingressConfig, componentRoute.Namespace, componentRoute.Name)
+		if existingComponentRoute != nil {
+			// Create a copy for compairison and remove transaction times
+			existingComponentRouteCopy := existingComponentRoute.DeepCopy()
+			setLastTransactionTime(existingComponentRouteCopy, now)
+
+			// Check if an update is needed
+			if equality.Semantic.DeepEqual(componentRouteCopy, existingComponentRouteCopy) {
+				return nil
+			}
+			*existingComponentRoute = *componentRoute
+		} else {
+			ingressConfig.Status.ComponentRoutes = append(ingressConfig.Status.ComponentRoutes, *componentRoute)
+		}
+
+		_, err = routeComponentHelper.ingressClient.UpdateStatus(ctx, ingressConfig, metav1.UpdateOptions{})
+		updated = err == nil
+		return err
+	})
+
+	return updated, err
+}
+
+func setLastTransactionTime(componentRoute *configv1.ComponentRouteStatus, now metav1.Time) {
+	for i := range componentRoute.Conditions {
+		componentRoute.Conditions[i].LastTransitionTime = now
+	}
+}
+
+// ConfigureRoute configures route.Spec.Host along with
```

Comment:
This is an incomplete sentence.

Reply:
After I moved the controller into the Library Go, this comment probably is no longer applicable. Resolving.